pkg/inspectors/signaturealgorithm/signaturealgorithm_test.go

```diff
@@ -0,0 +1,149 @@
+		publicKeyAlgorithm x509.PublicKeyAlgorithm
+		expectAllow        bool
+	}{
+		{x509.MD5WithRSA, x509.RSA, true},
+		{x509.SHA1WithRSA, x509.RSA, true},
+		{x509.SHA256WithRSA, x509.RSA, false},
+		{x509.SHA384WithRSA, x509.RSA, false},
+		{x509.SHA512WithRSA, x509.RSA, false},
+		{x509.SHA256WithRSAPSS, x509.RSA, false},
+		{x509.SHA384WithRSAPSS, x509.RSA, false},
+		{x509.SHA512WithRSAPSS, x509.RSA, false},
+		// dsa.PrivateKey doesn't implement crypto.Signer
+		// { x509.DSAWithSHA1, x509.DSA, false },
+		// { x509.DSAWithSHA256, x509.DSA, false },
+		{x509.ECDSAWithSHA1, x509.ECDSA, true},
+		{x509.ECDSAWithSHA256, x509.ECDSA, true},
+		{x509.ECDSAWithSHA384, x509.ECDSA, true},
+		{x509.ECDSAWithSHA512, x509.ECDSA, true},
+	} {
+		assertInspectionResult(t, inspector, testcase.signatureAlgorithm, testcase.publicKeyAlgorithm, testcase.expectAllow)
+	}
+}
+
+func TestConfigureBadAlgorithm(t *testing.T) {
+	for _, algorithm := range []string{
+		"MD2WithRSA",
+		"DSAWithSHA1",
+		"DSAWithSHA256",
+		"Unknown",
+	} {
+		t.Run(algorithm, func(t *testing.T) {
+			inspector, exists := inspectors.Get("signaturealgorithm")
+			require.True(t, exists, "inspectors.Get(\"signaturealgorithm\") to exist")
+
+			err := inspector.Configure(algorithm)
+			assert.EqualErrorf(t, err, fmt.Sprintf("unsupported SignatureAlgorithm %s", algorithm), algorithm)
+		})
+	}
+}
+
+func assertInspectionResult(t *testing.T, inspector inspectors.Inspector, signatureAlgorithm x509.SignatureAlgorithm, publicKeyAlgorithm x509.PublicKeyAlgorithm, expectAllow bool) {
+	t.Run(signatureAlgorithm.String(), func(t *testing.T) {
+		var key interface{}
+		var err error
+
+		switch publicKeyAlgorithm {
+		case x509.RSA:
+			// Generate a private key.
+			key, err = rsa.GenerateKey(rand.Reader, 1024)
+		case x509.ECDSA:
+			key, err = ecdsa.GenerateKey(elliptic.P224(), rand.Reader)
+		default:
+			t.Error("Cannot test key algorithm type", publicKeyAlgorithm)
+			return
+		}
+		require.NoError(t, err, "Generate the private key")
+
+		// Generate the certificate request.
+		certificateRequestTemplate := x509.CertificateRequest{
+			Subject: pkix.Name{
+				CommonName: "example.invalid",
+			},
+			SignatureAlgorithm: signatureAlgorithm,
+		}
+
+		certificateRequest, err := x509.CreateCertificateRequest(rand.Reader, &certificateRequestTemplate, key)
+		require.NoError(t, err, "Generate the CSR")
+
+		certificateRequestBytes := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE REQUEST", Bytes: certificateRequest})
+
+		request := certificates.CertificateSigningRequest{
+			Spec: certificates.CertificateSigningRequestSpec{
+				Username: "someRandomUser",
+				Request:  certificateRequestBytes,
+			},
+		}
+		message, err := inspector.Inspect(client, &request)
+		expectedMessage := ""
+		if !expectAllow {
+			expectedMessage = fmt.Sprintf("SignatureAlgorithm is %s", signatureAlgorithm)
+		}
+		assert.Equal(t, expectedMessage, message, "SignatureAlgorithm", signatureAlgorithm)
```

Comment:
I may be misunderstanding how the assert package works. But looking at the source code it looks like it does a Sprintf into the first argument after message passing all the remaining arguments.  If so, Shouldn't it be "SignatureAlgorithm %s", signatureAlgorithm ?

Reply:
Good catch. Pushed fixup commits.